eval_metrics.py

```diff
@@ -0,0 +1,62 @@
+from abstractmodel import AbstractLabelModel
+from dumb_model import DumbModel
+import fire
+from typing import TypedDict
+import pandas as pd
+import common_utils
+
+
+class ParsedDataset(TypedDict):
+    train: list[pd.DataFrame]
+    test: list[pd.DataFrame]
+    split: list[pd.DataFrame]
+
+
+def eval_model(model: AbstractLabelModel, parsed_dataset: ParsedDataset):
+    result = {"train": {}, "test": {}, "eval": {}}
+    for split_name, split in parsed_dataset.items():
+        count_cos_sim = []
+        for subgroup in split:
+            true_label = subgroup["category"][0]
+            texts = list(subgroup["title"])
+            generated_label = model.get_label(texts)
+            cos_sim = common_utils.embedder.encode(
+                true_label
+            ) @ common_utils.embedder.encode(generated_label)
+            count_cos_sim.append(cos_sim)
+        assert len(count_cos_sim) > 0, f"Length of {split_name} is 0"
+        average_cos_sim = sum(count_cos_sim) / len(count_cos_sim)
+        result[split_name]["avg_cos_sim"] = average_cos_sim
+    return result
+
+
+def read_diseases_dataset() -> ParsedDataset:
```

Comment:
I guess all datasets will look the same, so maybe let's implement a generic `read_dataset(dataset_name: str)` method instead